### PR TITLE
Fixes #6717 - Re-enable Screenshots tests

### LINKED ISSUE #6717: [Screenshot Tests] Re-enable these tests
Once the [PR to have screengrab working on Android X](https://github.com/fastlane/fastlane/pull/15217) landed on fastlane and they released a new version, it may be time to work on this again so that these tests are working correctly in case a set of new screenshots are needed.

### PR DESCRIPTION
**Pull request by isabelrios**
Fixes #6717 - This PR is to update the tests so that once the screenshots environment works again (with the next [fastlane release](https://github.com/fastlane/fastlane/releases)) the tests will pass too. 
Until then this work is needed to check that all tests are up to date with the latest implementation/robots and so.

---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by isabelrios** [comment]
Current screenshots taken by the tests (example running tests in fr-FR) can be found in [gcloud](https://console.cloud.google.com/storage/browser/l10n-fenix/2019-11-20/android/phoneScreenshots/?project=l10n-screenshots)

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt [hunk 1]
```diff
@@ -91,7 +88,7 @@ class HomeScreenRobot {
     }
     fun selectRenameCollection() {
         onView(allOf(ViewMatchers.withText("Rename collection"))).click()
-        mDevice.wait(Until.findObject(By.res("org.mozilla.fenix.debug:id/name_collection_edittext")), waitingTime)
+        mDevice.wait(Until.findObject(By.res("name_collection_edittext")), waitingTime)
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt]: should be: `mDevice.waitNotNull` ?

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt [hunk 2]
```diff
@@ -102,8 +99,9 @@ class HomeScreenRobot {
         mDevice.wait(Until.findObject(By.res("org.mozilla.fenix.debug:id/collections_header")), waitingTime)
     }
     fun typeCollectionName(name: String) {
-        mDevice.wait(Until.findObject(By.res("org.mozilla.fenix.debug:id/name_collection_edittext")), waitingTime)
-        collectionNameTextField().check(matches(hasFocus()))
+        mDevice.wait(Until.findObject(By.res("name_collection_edittext")), waitingTime)
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt]: same as above

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt [hunk 3]
```diff
@@ -102,8 +99,9 @@ class HomeScreenRobot {
...
+        // collectionNameTextField().check(matches(hasFocus()))
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt]: can remove if tests aren't flaky without it

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt [hunk 4]
```diff
@@ -194,3 +195,4 @@ private fun assertAboutFirefoxPreview() {
 }
 
 private fun goBackButton() = onView(CoreMatchers.allOf(ViewMatchers.withContentDescription("Navigate up")))
+fun swipeToBottom() = onView(ViewMatchers.withId(R.id.recycler_view)).perform(ViewActions.swipeUp())
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsRobot.kt]: per kotlin style guide, we should add 1 line between functions

File: app/src/androidTest/java/org/mozilla/fenix/ui/screenshots/MenuScreenShotTest.kt [hunk 5]
```diff
@@ -1,242 +1,206 @@
...
+        }.closeBrowserMenuToBrowser {
+        }.openHomeScreen {
+            Screengrab.screenshot("homescree-with-tabs")
+            closeTab()
+            Screengrab.screenshot("remove-tab")
+        }
+    }
+}
+
+fun openHistoryThreeDotMenu() = onView(withText(R.string.library_history)).click()
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/screenshots/MenuScreenShotTest.kt]: style add 1 blank line, same as above

File: app/src/androidTest/java/org/mozilla/fenix/ui/screenshots/ScreenshotTest.java [hunk 6]
```diff
@@ -1,75 +1,75 @@
...
+ import org.junit.runner.Description;
+
+ import tools.fastlane.screengrab.Screengrab;
+ import tools.fastlane.screengrab.UiAutomatorScreenshotStrategy;
+
+ /**
+  * Base class for tests that take screenshots.
+  */
+  public abstract class ScreenshotTest {
+     final long waitingTime = DateUtils.SECOND_IN_MILLIS * 10;
```
Comment (rpappalax) [inline app/src/androidTest/java/org/mozilla/fenix/ui/screenshots/ScreenshotTest.java]: perhaps move constant to TestHelper (or remove)

**Comment by isabelrios** [comment]
@rpappalax I have been running these tests for a few days several times a day and they look stable. Have not seen any failure. I think these tests could land so that they don't get out of date again until we have a way to run them separately as we discussed.

**Comment by isabelrios** [comment]
@rpappalax even though tests are running fine, I have changed them to a different package so that they don't run every time with the regular ui tests. This would be the first step to add a cron job to run them periodically (TBD how often) in a follow up PR, does that sound good?

**Comment by rpappalax** [comment]
> @rpappalax even though tests are running fine, I have changed them to a different package so that they don't run every time with the regular ui tests. This would be the first step to add a cron job to run them periodically (TBD how often) in a follow up PR, does that sound good?

Yes, let's do that.  Since they'll be in a different package (and will not be run on every github event), I assume for now you'd just be executing them manually as before? (just want to make sure they're getting run at some regular interval so they don't get stale again :)  Let's setup some time in a few weeks or early Jan. to put the cron in place.

**Comment by isabelrios** [comment]
> > @rpappalax even though tests are running fine, I have changed them to a different package so that they don't run every time with the regular ui tests. This would be the first step to add a cron job to run them periodically (TBD how often) in a follow up PR, does that sound good?
> 
> Yes, let's do that. Since they'll be in a different package (and will not be run on every github event), I assume for now you'd just be executing them manually as before? (just want to make sure they're getting run at some regular interval so they don't get stale again :) Let's setup some time in a few weeks or early Jan. to put the cron in place.

Yes, forgot to  add that, I will be running them at least once a week so that they keep up to date until we have that cron job. Thanks!

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt (hunk 2 above)
Comment (isabelrios) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/HomeScreenRobot.kt]: for some reason that I don't know the tests do not work with `mDevice.waitNotNull` that's why I left this one this way.

**Pull request merged**